components/nexus-repository/src/test/java/org/sonatype/nexus/repository/storage/StorageFacetImplIT.java

```diff
@@ -561,4 +604,160 @@ public void duplicateAssetName() throws Exception {
+      assertThat(asset.formatAttributes().get("attribute1", String.class), equalTo("original"));
+    }
+  }
+
+  @Test
+  public void transactionsRollBackWhenRequired() throws Exception {
+    try (StorageTx tx = underTest.openTx()) {
+      final Component component = tx.createComponent(tx.getBucket(), testFormat)
+          .group("myGroup")
+          .version("0.9")
+          .name("myComponent");
+      tx.saveComponent(component);
+      tx.rollback();
+    }
+
+    try (StorageTx tx = underTest.openTx()) {
+      final Component component = tx.findComponentWithProperty("group", "myGroup", tx.getBucket());
+      assertThat(component, is(nullValue()));
+    }
+  }
+
+  @Test
+  public void transactionsRollBackOnException() throws Exception {
+    try {
+      try (StorageTx tx = underTest.openTx()) {
+        final Component component = tx.createComponent(tx.getBucket(), testFormat)
+            .group("myGroup")
+            .version("0.9")
+            .name("myComponent");
+        tx.saveComponent(component);
+        if (true) {
+          throw new MidTransactionException();
+        }
+      }
+    }
+    catch (MidTransactionException ignored) {
+    }
+
+    try (StorageTx tx = underTest.openTx()) {
+      final Component component = tx.findComponentWithProperty("group", "myGroup", tx.getBucket());
+      assertThat(component, is(nullValue()));
+    }
+  }
+
+  @Test
+  public void transactionContentIsSaved() throws Exception {
+    try (StorageTx tx = underTest.openTx()) {
+      final Component component = tx.createComponent(tx.getBucket(), testFormat)
+          .group("myGroup")
+          .version("0.9")
+          .name("myComponent");
+      tx.saveComponent(component);
+      tx.commit();
+    }
+
+    try (StorageTx tx = underTest.openTx()) {
+      final Iterable<Component> components = tx.browseComponents(tx.getBucket());
+      for (Component c : components) {
+        System.err.println(c);
+      }
+
+      final Component component = tx.findComponentWithProperty("group", "myGroup", tx.getBucket());
+      assertThat(component, is(notNullValue()));
+      assertThat(component.group(), is("myGroup"));
+    }
+  }
+
+  @Test
+  public void entityIdCanBeUsedInLaterTransactions() throws Exception {
+    EntityId componentId = null;
+    EntityId assetId = null;
+
+    try (StorageTx tx = underTest.openTx()) {
+      final Component component = tx.createComponent(tx.getBucket(), testFormat).name("component");
+      tx.saveComponent(component);
+
+      componentId = id(component);
+
+      tx.commit();
+    }
+
+    try (StorageTx tx = underTest.openTx()) {
+      final Component component = tx.findComponent(componentId, tx.getBucket());
+      assertThat("component", component, is(notNullValue()));
+      assertThat(component.name(), is("component"));
+    }
+  }
+
+  @Test
+  public void multipleEntityIdsCanBeUsedInLaterTransactions() throws Exception {
```

Comment:
this covers the previous test conditions completely afaict, might as well just rename to 'entityIdCanBeUsedInLaterTransactions' and delete that present implementation